sormas-ui/src/main/java/de/symeda/sormas/ui/visit/VisitEditForm.java

```diff
@@ -124,6 +140,37 @@ public void validate(Object value) throws InvalidValueException {
 			});
 		}
 
+
+		if (caze != null) {
+			getField(VisitDto.VISIT_DATE_TIME).addValidator(new Validator() {
+
+				private static final long serialVersionUID = 6720804629542567720L;
+
+				@Override
+				public void validate(Object value) throws InvalidValueException {
+					Date visitDateTime = (Date) getFieldGroup().getField(VisitDto.VISIT_DATE_TIME).getValue();
+					Date startDate = CaseLogic.getStartDate(caze.getSymptoms().getOnsetDate(), caze.getReportDate());
+					if (visitDateTime.before(startDate)
+							&& DateHelper.getDaysBetween(visitDateTime, caze.getReportDate()) > VisitDto.ALLOWED_CONTACT_DATE_OFFSET) {
+						if (caze.getSymptoms().getOnsetDate() != null) {
+							throw new InvalidValueException(
+									I18nProperties.getValidationError(Validations.visitBeforeSymptomsOnSet, VisitDto.ALLOWED_CONTACT_DATE_OFFSET));
+						} else {
+							throw new InvalidValueException(
+									I18nProperties.getValidationError(Validations.visitBeforeCaseReport, VisitDto.ALLOWED_CONTACT_DATE_OFFSET));
+						}
+					}
+					Date endDate = CaseLogic.getEndDate(caze.getReportDate(), caze.getFollowUpUntil());
+					if (endDate != null
+							&& visitDateTime.after(endDate)
+							&& DateHelper.getDaysBetween(endDate, visitDateTime) > VisitDto.ALLOWED_CONTACT_DATE_OFFSET) {
+						throw new InvalidValueException(
+								I18nProperties.getValidationError(Validations.visitAfterFollowUp, VisitDto.ALLOWED_CONTACT_DATE_OFFSET));
+					}
+				}
+			});
+		}
```

Comment:
I think this whole code block could be unified with the one for contact != null above by introducing some local variables to store the relevant dates before executing the logic

Reply:
Why not.
Since the messages and values are different, I'm not sure it's more readable.

Reply:
Yes, but the code apart from that is exactly identical, which means that by introducing variables like "startDate", "endDate", "errorMessageDateTooEarly" and "errorMessageDateTooLate", a lot of duplicate code lines can be removed.